website/source/docs/docker/networking.html.md

```diff
@@ -0,0 +1,222 @@
+---
+layout: "docs"
+page_title: "Networking - Docker Provider"
+sidebar_current: "providers-docker-networking"
+description: |-
+  The Vagrant Docker provider supports using the private network using the
+  `docker network` commands.
+---
+
+# Networking
+
+Vagrant uses the `docker network` command under the hood to create and manage
+networks for containers. Vagrant will do its best to create and manage networks
+for any containers configured inside the Vagrantfile. Each docker network is grouped
+by the subnet used for a requested ip address.
+
+For each newly unique network, Vagrant will run the `docker network create` subcommand
+with the provided options from the network config inside your Vagrantfile. If multiple
+networks share the same subnet, Vagrant will reuse that existing network for multiple
+containers. Once these networks have been created, Vagrant will attach these
+networks to the requested containers using the `docker network connect` for each
+network.
+
+Vagrant names the networks inside docker as `vagrant_network` or `vagrant_network_<subnet here>`
+where `<subnet_here>` is the subnet for the network if defined by the user. An
+example of these networks is shown later in this page. If no subnet is requested
+for the network, Vagrant will connect the `vagrant_network` to the container.
+
+When destroying containers through Vagrant, Vagrant will clean up the network if
+there are no more containers using the network.
+
+## Docker Network Options
+
+Most of the options given align with the command line flags
+for the [docker network create](https://docs.docker.com/engine/reference/commandline/network_create/)
+command. If there are any specific options you want to enable from the `docker network create`
+command, you can specify them like this:
+
+```ruby
+docker.vm.network :private_network, type: "dhcp", docker_network__internal: true
+```
+
+This will enable the `internal` option for the network when created with `docker network create`.
+
+Where `option` corresponds to the given flag that will be provided to the `docker network create`
+command. Similarly, if there is a value you wish to enable when connecting a container
+to a given network, you can use the following value in your network config:
+
+```ruby
+docker_connect__option: "value"
+```
+
+It should be noted that if you want the container to have a specific IP instead
+of using DHCP, you also will have to specify a subnet due to how docker networks behave.
```

Comment:
Defining the netmask/subnet is not required. If undefined, it will default to `/24` for ipv4 and `/64` for ipv6. If a network already exists and the provided IP address is within that network's subnet, that network will be used.